### PR TITLE
Fix stylelint errors in platform directory

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

This should not be merged until https://github.com/department-of-veterans-affairs/vets-website/pull/17305 gets merged. This fixes the `stylelint` errors that that PR revealed in the `platform` directory.

## Testing done

I ran `stylelint` while on the branch for #17305 and fixed the issues that way.

## Screenshots

What we are fixing: 

image

After:

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/sass/modules/_m-medallia.scss [hunk 1]
```diff
@@ -107,5 +107,5 @@
   top: 40px !important;
   right: 12px !important;
   z-index: 9999999;
-  color: #9f9fa4 !important;
+  color: $color-gray-light !important;
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/modules/_m-medallia.scss:110]: image
image

`color-gray-light` seemed like the closest color from https://design.va.gov/design/color-palette

File: src/platform/site-wide/sass/modules/_m-responsive-table.scss [hunk 2]
```diff
@@ -18,7 +16,7 @@ $color-responsive-table-border: #999;
   }
   .va-table--responsive {
     tr {
-      border-top: 2px solid $color-responsive-table-border;
+      border-top: 2px solid $color-gray-light;
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/modules/_m-responsive-table.scss:19]: image
image


This one isn't _as_ close

File: src/platform/site-wide/sass/modules/_m-side-nav.scss [hunk 3]
```diff
@@ -1,5 +1,3 @@
-$side-nav-color-gray-lightest: #f8f8f8; // I could not find the color in our design utilities.
-$side-nav-color-dark: #323a45;
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/modules/_m-side-nav.scss:2]: `#323a45` is already `$color-gray-dark`, so no color matching needed.

File: src/platform/site-wide/sass/modules/_m-side-nav.scss [hunk 4]
```diff
@@ -1,5 +1,3 @@
-$side-nav-color-gray-lightest: #f8f8f8; // I could not find the color in our design utilities.
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/modules/_m-side-nav.scss:1]: image
image


This is pretty close to `$color-gray-lightest`

File: src/platform/site-wide/sass/modules/_m-side-nav.scss [hunk 5]
```diff
@@ -172,8 +175,8 @@ $side-nav-color-dark: #323a45;
     .va-sidenav-level-1 {
       margin-top: 2.3em;
       &:first-child {
-        margin-top: .75em; // the wrapper already has top padding, so we don't need as much space here
-        @media(max-width: $medium-screen) {
+        margin-top: 0.75em; // the wrapper already has top padding, so we don't need as much space here
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/modules/_m-side-nav.scss:178]: These other changes are what happened on save.

File: src/platform/site-wide/sass/shame.scss [hunk 6]
```diff
@@ -41,7 +41,6 @@ $medium-phone-screen: 375px;
   #va-header-logo-menu {
     align-items: center;
   }
-  $color-primary-lightest: #e1f3f8;
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/shame.scss:44]: This wasn't used, but it is ` $color-primary-alt-lightest `

File: src/platform/site-wide/sass/shame.scss [hunk 7]
```diff
@@ -592,7 +591,7 @@ img[data-srcset] {
   }
 }
 ._acs._acsbadge--default {
-  background-color: #000 !important;
+  background-color: $color-black !important;
```
Comment (bkjohnson) [inline src/platform/site-wide/sass/shame.scss:594]: This variable matches the hex code.

**Review by bkjohnson — COMMENTED** [review]
Pointing out where I had to guess at colors :shrug:

**Review by jhonny218 — APPROVED** [review]
LGTM

**Comment by mpelzsherman** [comment]
The code changes look fine, but we should have a designer (e.g. @lildziner) sign off on the color changes.

In theory it should be better to use the design system colors, but I don't think we can safely assume the colors chosen here will be acceptable for design. We should also make sure the color changes don't introduce a11y issues due to insufficient contrast.

**Comment by rtwell** [comment]
@bkjohnson is correct — we must use colors specified in the design system. if these changes introduce any contrast issues (at first glance, they should not), the team can use other colors provided in the design system to address.

**Review by bkjohnson — COMMENTED** [review]
Here is some more context to the color changes.

## Background of selected anchor tag

### Currently

image

### Change to f1f1f1

image

## Background of `.open` node

Within the sidenav it looks like the only time the `open` class is applied is when there is also the `va-sidenav-item-label` class:

https://github.com/department-of-veterans-affairs/vets-website/blob/2e606ebf0cbc017f8f2d94686b7681f3c9d17742/src/platform/site-wide/side-nav/components/NavItemRow.js#L62-L68

https://github.com/department-of-veterans-affairs/vets-website/blob/2e606ebf0cbc017f8f2d94686b7681f3c9d17742/src/platform/site-wide/side-nav/components/NavItemRow.js#L94-L96

I haven't found an example of this in production, but it's the same styling change we see with the `selected` node.

---

## Responsive table changes

https://www.va.gov/disability/compensation-rates/special-monthly-compensation-rates/past-rates-2020/

### Currently

image

### Color of top border changed

image


---

I'm having a hard time finding examples of the medallia styles on a production page, so no in-context screenshots for those color changes

**Review by mpelzsherman — APPROVED** [review]

**Pull request merged**